<h1><b>Tarea 2: Reporte de incertidumbre <b></h1>

---
<div align="right">
<font size=2>
Estiven Castrillón Alzate<br>
Institute of Physics<br>
</font>
</div>

1. Medición del largo del péndulo.

¿Vale la pena hacer mediciones repetitivas de la variable o si por el contrario con una es suficiente? Para efectos prácticos de la actividad no se tomaron varias medidas de la longitud del péndulo, pero lo ideal sería que sí, esto para hacer un reporte de incertidumbre más completo de la aceleración de la gravedad estimada a partir de los datos de periodo y largo del péndulo, esto debido a la dependencia de la gravedad de la longitud y el periodo pendular, así como se muestra en el siguiente proceso:
\begin{equation}
    T = 2\pi \sqrt{\frac{l}{g}} \tag{1}
\end{equation}
Donde T es el período pendular, g es la aceleración de la gravedad y l es la longitud del péndulo, luego:
\begin{equation}
    g = \frac{4 l\pi^2}{T^2} \tag{2}
\end{equation}

La longitud del péndulo es $ (73.6 \pm 0.1) \space \text{cm}$

2. Medición del periodo de oscilación pendular.
Es preferible que mida el tiempo de varios periodos y luego divida, ¿por qué?.

Ya que como es un ángulo tan pequeño y la longitud de la cuerda no es muy grande, las oscilaciones son bastante rápidas, cosa que puede ser contraproducente a la hora de medir el tiempo con un cronómetro; así, cuando se cronometran varios periodos hay una menor probabilidad de error a la hora de empezar y detener el cronómetro.

In [209]:
import pandas as pd
import numpy as np

from math import radians

Importación de datos

In [210]:
# # #Importación de datos mediante archivos en la máquina local
# datos_oscilaciones = pd.read_excel('2.0-datos_oscilaciones_péndulo.xlsx')

# Importación de datos con base en un archivo json, (no necesita tener el archivo de datos en la máquina local)
datos_oscilaciones_json = '{"1_oscilacion(s)":{"0":1.58,"1":1.65,"2":1.54,"3":1.63,"4":1.61},"5_oscilaciones(s)":{"0":8.73,"1":8.77,"2":8.84,"3":8.81,"4":8.81},"7_oscilaciones(s)":{"0":12.21,"1":12.25,"2":12.24,"3":12.26,"4":12.16},"9_oscilaciones(s)":{"0":15.82,"1":15.98,"2":15.92,"3":15.93,"4":15.9},"10_oscilaciones(s)":{"0":17.69,"1":17.46,"2":17.6,"3":17.56,"4":17.57}}'
datos_oscilaciones = pd.read_json(datos_oscilaciones_json)
datos_oscilaciones

1_oscilacion(s)  5_oscilaciones(s)  7_oscilaciones(s)  9_oscilaciones(s)  \
0             1.58               8.73              12.21              15.82   
1             1.65               8.77              12.25              15.98   
2             1.54               8.84              12.24              15.92   
3             1.63               8.81              12.26              15.93   
4             1.61               8.81              12.16              15.90   

   10_oscilaciones(s)  
0               17.69  
1               17.46  
2               17.60  
3               17.56  
4               17.57

Dividiendo cada valor de la columna por el numero de oscilaciones para obtener el periodo


In [211]:
periodos_oscilacion = datos_oscilaciones.copy() # Creando una copia de los datos originales

for columna in periodos_oscilacion:
    cantidad_oscilaciones = int(columna.split('_')[0])
    periodos_oscilacion[columna] = round(periodos_oscilacion[columna] / cantidad_oscilaciones, 2)

periodos_oscilacion

1_oscilacion(s)  5_oscilaciones(s)  7_oscilaciones(s)  9_oscilaciones(s)  \
0             1.58               1.75               1.74               1.76   
1             1.65               1.75               1.75               1.78   
2             1.54               1.77               1.75               1.77   
3             1.63               1.76               1.75               1.77   
4             1.61               1.76               1.74               1.77   

   10_oscilaciones(s)  
0                1.77  
1                1.75  
2                1.76  
3                1.76  
4                1.76

In [212]:
print('-----Promedio de cada columna-----')
promedios_periodos = round(periodos_oscilacion.mean(), 2)
promedios_periodos

-----Promedio de cada columna-----


1_oscilacion(s)       1.60
5_oscilaciones(s)     1.76
7_oscilaciones(s)     1.75
9_oscilaciones(s)     1.77
10_oscilaciones(s)    1.76
dtype: float64

De igual forma, discuta si vale la pena hacer mediciones repetitivas de la variable o si por el
contrario con una basta. ¿Qué estrategia debería usar para disminuir el error provocado por
la manipulación humana del cronómetro?

* Realizar mediciones repetitivas es crucial para evaluar la consistencia y precisión de los resultados. A través de mediciones repetidas, es posible calcular promedios y desviaciones estándar, lo que brinda una idea más precisa de la magnitud de la incertidumbre asociada con las mediciones. Cuantas más mediciones repetitivas se realicen, mejor será la estimación de la verdadera magnitud de la variable y su incertidumbre.

* Primeramente, que la medición del tiempo la realice la misma persona que está poniendo el péndulo a una apertura de 15 grados, esto para minimizar el error asociado a la respuesta cerebral a la hora de cronometrar y escuchar la señal de que ya ha dejado en libertad al péndulo. Segundo, optimizar el angulo de visión de la persona que esta midiendo el tiempo con respecto al pendulo y asi evitar que pare el tiempo antes o despues por un mal angulo de perspectiva. Tercero, tener las mejores condiciones ambietales posibles, asegurarnos de tener un espacio con la menor cantidad de corriente de aire posible, asi evitar que la friccion con el viento altere nuestras medidas, de igual manera tener una superficie estable, que no se vea perturbada por el movimiento del pendulo, ya que en caso de que la superficie se mueva por culpa del pendulo la medida se vera afectada.



Reporte adecuadamente T con su respectiva incertidumbre. Considere que debe estimar una incertidumbre total que de cuenta tanto de la apreciación del instrumento usado como de la incertidumbre por fluctuaciones aleatorias.

Construya una tabla que contenga para un número dado de medidas columnas con el valor esperado, la desviación estándar, el cuadrado de la incertidumbre estándar, el cuadrado de la mitad de la resolución del instrumento, la suma en cuadratura de la incertidumbre estándar y la resolución del instrumento. Realice esto para diferente número de medidas. Comente sobre el comportamiento de los valores en las diferentes columnas.

Reporte de T, (periodo)

In [213]:
# Desviación estándar muestral
periodos_array = periodos_oscilacion.values.flatten()
desviacion_est = periodos_array.std()
media_periodos = periodos_array.mean()

# Incertidumbre estándar de la media
incert_estandar = desviacion_est / np.sqrt(len(periodos_array))

# Incertidumbre del instrumento de medición, (cronómetro)
incert_cronometro = 0.01

# Incertidumbre total
incert_total = np.sqrt( (incert_estandar ** 2) + (incert_cronometro ** 2) )

# Reporte de T con incertidumbre
print(f'T_periodo = ({round(media_periodos, 2)} ± {round(incert_total, 2)}) segundos')

T_periodo = (1.73 ± 0.02) segundos


Tabla de medidas estadísiticas

In [214]:
periodos_array = periodos_oscilacion.values.flatten()

def tabla_estadistica_muestras(tamaños_muestras):
    """
    Calcula medidas estadísticas para una muestra de datos.

    Parameters:
        tamaños_muestras (list): Lista de tamaños de muestra a considerar.

    Returns:
        pd.DataFrame: Un DataFrame con medidas estadísticas para cada tamaño de muestra.
    """
    incert_instrumento = 0.01

    resultados = []
    for tamaño_muestra in tamaños_muestras:
        # Tomando la muestra de los datos
        muestra_datos = periodos_array[:tamaño_muestra]

        media_datos = np.round(muestra_datos.mean(), 2)
        desviacion_datos = np.round(muestra_datos.std(), 2)

        # Incertidumbre estándar al cuadrado
        incert_est_cuadrado = np.round((desviacion_datos / np.sqrt(tamaño_muestra)) ** 2, 4)

        # Cuadrado de la mitad de la resolución del instrumento
        incert_mitad_instrumento = np.round((incert_instrumento / 2) ** 2, 5)

        # Suma en cuadratura de la incertidumbre estándar y la resolución del instrumento
        incert_total_muestra = np.round(np.sqrt(incert_est_cuadrado + (incert_instrumento ** 2)), 2)

        # Definición de la tabla con medidas estadísticas
        resultados.append({
            'Tamaño de Muestra': tamaño_muestra,
            'Media': media_datos,
            'Desviación Estándar': desviacion_datos,
            'Incertidumbre Estándar Cuadrado': incert_est_cuadrado,
            'Incertidumbre Mitad Instrumento Cuadrado': incert_mitad_instrumento,
            'Incertidumbre Total': incert_total_muestra
        })

    df_resultados = pd.DataFrame(resultados)
    return df_resultados

In [215]:
tabla_estadistica_muestras(tamaños_muestras=[5, 10, 15, 25, 30])

Tamaño de Muestra  Media  Desviación Estándar  \
0                  5   1.72                 0.07   
1                 10   1.73                 0.06   
2                 15   1.72                 0.07   
3                 25   1.73                 0.07   
4                 30   1.73                 0.07   

   Incertidumbre Estándar Cuadrado  Incertidumbre Mitad Instrumento Cuadrado  \
0                           0.0010                                   0.00002   
1                           0.0004                                   0.00002   
2                           0.0003                                   0.00002   
3                           0.0002                                   0.00002   
4                           0.0002                                   0.00002   

   Incertidumbre Total  
0                 0.03  
1                 0.02  
2                 0.02  
3                 0.02  
4                 0.02

- **Conclusiones de la tabla estadísica:** A medida que el tamaño de la muestra aumenta, tanto la media como la incertidumbre estándar al cuadrado y la incertidumbre total tienden a disminuir, por otro lado, la desviación estándar de la muestra aumenta. Sin embargo, los cambios en estas medidas son muy pequeños, aun así, todos tienden a acercarse al valor estimado de la medida con incertidumbre total mencionado en el punto anterior, el cual fue de $T = (1.73 \pm 0.02) \space \text{segundos}$

3. Determinación de g:

a) Use las mediciones directas de l y T para calcular el valor de la aceleración gravitacional en el
laboratorio. Reporte su resultado haciendo el análisis de la incertidumbre por dos métodos:

- Teoría elemental de la incertidumbre

Utilizando la ecuación (2) para calcular la gravedad con la longitud del péndulo y su periodo calculado en el punto anterior, es decir, los valores más probables de longitud del péndulo y periodo

In [216]:
#Definimos los valores para hallar g

longitud_pendulo = 0.736 # Longitud en metros
periodo_pendulo = 1.73 # Valor del perído calculado en el punto anterior

# Calculando aceleración de gravedad

gravedad_calculada = (4 * (np.pi ** 2) * longitud_pendulo) / periodo_pendulo ** 2

print(f'Gravedad = {round(gravedad_calculada, 2)} metros/segundo^2')

Gravedad = 9.71 metros/segundo^2


Incertidumbre analizando valores extremos

In [217]:
gravedad_minima = (4 * (np.pi ** 2) * longitud_pendulo) / (promedios_periodos.max().max() ** 2)
gravedad_maxima = (4 * (np.pi ** 2) * longitud_pendulo) / (promedios_periodos.min().min() ** 2)

incert_teoria_elemental = round((gravedad_maxima - gravedad_minima) / 2, 2)
print(f'Incertidumbre por teoria elemental analizando valores extremos: {incert_teoria_elemental}')

Incertidumbre por teoria elemental analizando valores extremos: 1.04


Así, la gravedad por teoría elemental de la incertidumbre quedaría:
$$ g = (9.71 \pm 1.02)\space \frac{m}{s^2}$$

- Usando propagación por derivadas parciales de incertidumbres pequeñas.

$$\Delta g = \frac{\partial g}{\partial l}\Big |_l \cdot \Delta l + \frac{\partial g}{\partial T} \Big |_T \cdot \Delta T$$

así: $$\Delta g = \frac{4\pi^2}{(1.73 s)^2}\cdot(0.001 m) - \frac{2\pi^2(0.736 m)}{(1.73 s)} \cdot(0.02s) = 0.15 \frac{m}{s^2}$$

Este error por derivadas parciales es mucho menos al obtenido en el método anterior por teoría elemental de la incertidumbre, esto debido a que en el proceso de medición hubo algunos errores lo cual hizo que el análisis de valores extremos en el literal anterior tuviera valores muy grandes debido a estas grandes diferencias. En conclusión, el método de derivadas parciales es más preciso al no tener en cuenta errores procedimentales tan significativos como los outliers de las mediciones.

b) ¿Podría afirmar que una de las variables presenta una incertidumbre dominante respecto a
la otra en el contexto del análisis y propagación de la incertidumbre?

La influencia del factor tiempo genera una incertidumbre significativamente mayor en comparación con la que proviene de la longitud, debido a que la magnitud de la expresión $ \frac{\partial g}{\partial T} \Big |_T \cdot \Delta T$ es significativamente mayor.

c) Calculando la gravedad en Medellín y el error relativo respecto a la medida calculada en el experimento del péndulo

In [219]:
latitud = radians(6.2)
altitud_sobre_mar = 1464
A, B = 0.0053024, 0.0000058
valor_esperado_gravedad = 9.780327 * (1 + (A * latitud) - (B * 2 * latitud)) - (3.086e-6 * altitud_sobre_mar)

deltaX = (abs(valor_esperado_gravedad - gravedad_calculada) / valor_esperado_gravedad) * 100
print(f'El valor de la gravedad esperada es de: {round(valor_esperado_gravedad, 2)} metros/segundo^2')
print(f"Error relativo entre la medida esperada y la medida calculada: {round(deltaX, 2)}%")

El valor de la gravedad esperada es de: 9.78 metros/segundo^2
Error relativo entre la medida esperada y la medida calculada: 0.75%


En general, se hizo una muy buena aproximación de la gravedad con el experimento del péndulo, el error sistemático en la medida de los periodos fue el causante de algunas discrepancias entre la medida calculada y la medida esperada, aun así, se hizo una aproximación muy representativa de la realidad física.